executor/join_test.go

```diff
@@ -1599,9 +1599,9 @@ func (s *testSuiteJoin1) TestIssue13177(c *C) {
 	tk.MustExec("insert into t1 values(\"abcd\", 1, 1), (\"bacd\", 2, 2), (\"cbad\", 3, 3)")
 	tk.MustExec("insert into t2 values(\"bcd\", 1, 1), (\"acd\", 2, 2), (\"bad\", 3, 3)")
 	tk.MustQuery("select /*+ inl_merge_join(t1, t2) */ * from t1 join t2 on substr(t1.a, 2, 4) = t2.a and t1.b = t2.b where t1.c between 1 and 5").Check(testkit.Rows(
-		"abcd 1 1 bcd 1 1",
 		"bacd 2 2 acd 2 2",
 		"cbad 3 3 bad 3 3",
+		"abcd 1 1 bcd 1 1",
```

Comment:
why this line changed?

Reply:
It keeps order for column `acd bad bcd` instead of `abcd bacd cbad`. The reason why previous CI could be accepted is that I use `tidb_inl` to choose an index join before and after **Huaiyu**'s PR, the `tidb_inl` can't choose the index merge join.

Reply:
Will this result be stable enough?

Reply:
Always stable, index merge join keeps outer order.